shell/common/rasterizer.cc

```diff
@@ -474,6 +470,11 @@ RasterStatus Rasterizer::DrawToSurface(flutter::LayerTree& layer_tree) {
       surface_->GetContext()->performDeferredCleanup(kSkiaCleanupExpiration);
     }
 
+    // Clean the render context after submitting the frame.
+    // This ensures that the GL context is released after drawing to the
+    // surface.
+    surface_->ClearRenderContext();
```

Comment:
Why was the null ptr check removed here?

Reply:
`surface_ ` is guaranteed to be non-null at this point.

Reply:
Then why does the previous if statement also check it?

Reply:
It looks like `Rasterizer::Draw` doesn't check for `surface_` whereas `Rasterizer::DrawToSurface` always assumes that a surface exists; hence the name of the method.

Reply:
`::Draw` also calls `::DoDraw` and the latter method checks `surface_` for nullptr. It is checked by a number of control flows in the `::Draw` method so how much confidence do we have that it is never called with a null surface?

This PR is supposed to unblock the roll, it would be better to avoid potential incidental issues rather than making changes that "look good" and aren't directly related to fixing the critical issue at hand.

Reply:
There's also a `FML_DCHECK(surface_);` in https://github.com/flutter/engine/pull/20931/files#diff-e81d85a84b4999177209791f247205d4R407

Reply:
I think the case of calling `DrawToSurface` if `surface_` is `null` is covered by:

```c
RasterStatus Rasterizer::DrawToSurface(flutter::LayerTree& layer_tree) {
  TRACE_EVENT0("flutter", "Rasterizer::DrawToSurface");
  FML_DCHECK(surface_);
```

Reply:
DrawToSurface is called from DoDraw which protects it from null.

Draw calls DrawToSurface. None of that code is protecting it from having a null surface pointer here.

Reply:
I added the check in https://github.com/flutter/engine/pull/20931/commits/b6d72c38b2d804ddf0ee525124e197142dc77b3b

Reply:
MANY APOLOGIES!

I completely misread the original diff. It looked like the call to ClearRenderContext had been moved only a few lines down in the Draw() call, but I missed the "diff gap" where it was moved over 200 lines to a completely different method.

Please ignore my comments and now I see that it was always in a null-safe location.

Reply:
I'm going to resolve the conversation as a result of my misunderstanding, and so the DCHECK change can be reverted at some point. Apologies for the extra work and the confusing conversation...